lib/late_ack_tracker.c

```diff
@@ -141,7 +141,10 @@ late_ack_tracker_manage_msg_ack(AckTracker *s, LogMessage *msg, AckType ack_type
         _drop_range(self, ack_range_length);
 
         if (ack_type == AT_SUSPENDED)
-          log_source_flow_control_suspend(self->super.source);
+          {
+            log_source_flow_control_adjust(self->super.source, 1);
+            log_source_flow_control_suspend(self->super.source);
+          }
```

Comment:
I think the original code is not correct.
We should suspend immediately when an `AT_SUSPENDED` ack received even if we don't have continuous (acked=true) block.
So we should mark the ringbuffer record as acked, then call suspend.
The adjustment is not required here as the ringbuffer size is not decreasing.